rubrics/impl/src/main/java/org/sakaiproject/rubrics/impl/RubricsServiceImpl.java

```diff
@@ -876,11 +870,19 @@ public Optional<ToolItemRubricAssociation> saveRubricAssociation(String toolId,
                             }
                         }
                     }
+                } else if (requestedRubricId.equals(0L)){   //deactivating an association without making a new one
+                    association.setActive(false);
+                    associationRepository.save(association);
+                    return Optional.empty();
+                }
+            } else {    //if existingAssociation is not present, it could just mean that it was deactivated previously; the specific getRubricAssociation impl that we used earlier to load it will ignore deactivated ones.
+                Optional<ToolItemRubricAssociation> optionalExistingAssociation = findAssociationByItemIdAndRubricId(toolItemId, requestedRubricId);    // this will include inactive [soft-deleted] ones
+                if (optionalExistingAssociation.isPresent()) {  // if there's already an old association for the requested rubric that was deactivated previously, reuse it
+                    optionalExistingAssociation.get().setActive(true);
+                    return Optional.of(associationRepository.save(optionalExistingAssociation.get()));
                 }
-            } else {
-                // first association for this rubric
                 Optional<ToolItemRubricAssociation> newAssociation = createToolItemRubricAssociation(toolId, toolItemId, params, requestedRubricId);
-                if (newAssociation.isPresent()) {
+                if (newAssociation.isPresent()) {   //otherwise take this new association and save it.
```

Comment:
You don't really need this comment, I don't think. The code is pretty self explanatory, imo.